### PR TITLE
TS-19-v2-23: Refactor ESLint errors in services directory

### LINKED ISSUE #23: Refactor: ESLint errors in services, validation directories
Acceptance Criteria:
Fix ESLint errors in services, validation directories and App.js, index.js, serupTests.js files.

Estimate: 2 day

### PR DESCRIPTION
**Pull request by ViktoriiaKosovan**

### CODE REVIEW

File: frontend/src/components/ScheduleDialog/ScheduleDialog.js [hunk 1]
```diff
@@ -112,7 +112,7 @@ const ScheduleDialog = (props) => {
                                 renderInput={(params) => (
                                     <TextField
                                         {...params}
-                                        label={translation('formElements:room_label')}
+                                        label={translation(ROOM_LABEL)}
```
Comment (andrii-kuzmichov) [inline frontend/src/components/ScheduleDialog/ScheduleDialog.js:115]: make the same name for `translation` => `t` throughout the project

File: frontend/src/constants/services.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+export const BACK_END_SUCCESS_OPERATION = 'serviceMessages:back_end_success_operation';
```
Comment (andrii-kuzmichov) [inline frontend/src/constants/services.js:1]: Maybe create new file for constants related translation labels

Comment (andrii-kuzmichov) [reply frontend/src/constants/services.js:1]: and collect all labels in this file

File: frontend/src/containers/TeachersList/TeachersList.js [hunk 3]
```diff
@@ -252,9 +253,7 @@ const TeacherList = (props) => {
                 </aside>
 
                 <section className="container-flex-wrap">
-                    {visibleItems.length === 0 && (
-                        <NotFound name={t('formElements:teacher_a_label')} />
-                    )}
+                    {visibleItems.length === 0 && <NotFound name={t(TEACHER_LABEL)} />}
```
Comment (andrii-kuzmichov) [inline frontend/src/containers/TeachersList/TeachersList.js:256]: could you use `isEmpty` from lodash?

File: frontend/src/services/authService.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ import {
 } from '../constants/axios';
 
 export const authUserService = (request) => {
-    if (request.result.hasOwnProperty('authType') && request.result.authType === 'google') {
+    if (get(request.result, 'authType') && request.result.authType === 'google') {
```
Comment (andrii-kuzmichov) [inline frontend/src/services/authService.js:12]: `'google'` to constant

File: frontend/src/services/roomTypesService.js [hunk 5]
```diff
@@ -74,17 +68,23 @@ export const postNewType = (values) => {
...
 };
 
+export const addNewTypeService = (values) => {
+    if (values.id) {
+        putNewType(values);
+    } else {
+        postNewType(values);
+    }
+};
+
```
Comment (andrii-kuzmichov) [inline frontend/src/services/roomTypesService.js:87]: maybe create unified method for the same template in app

File: frontend/src/services/scheduleService.js [hunk 6]
```diff
@@ -202,33 +192,75 @@ export const deleteItemFromScheduleService = (itemId) => {
...
-        (!values.hasOwnProperty('group') &&
-            values.hasOwnProperty('teacher') &&
-            +values.teacher === 0) ||
-        (values.hasOwnProperty('group') &&
-            values.hasOwnProperty('teacher') &&
+        (!get(values, 'group') && !get(values, 'teacher')) ||
+        (get(values, 'group') && !get(values, 'teacher') && +values.group === 0) ||
+        (!get(values, 'group') && get(values, 'teacher') && +values.teacher === 0) ||
+        (get(values, 'group') &&
+            get(values, 'teacher') &&
```
Comment (andrii-kuzmichov) [inline frontend/src/services/scheduleService.js:263]: need refactor

File: frontend/src/services/uploadFile.js [hunk 7]
```diff
@@ -21,5 +21,5 @@ export const uploadStudentsToGroupFile = (file, groupId) => {
                 }),
             );
         })
-        .catch((err) => alert('File Upload Error'));
+        .catch(() => alert('File Upload Error'));
```
Comment (andrii-kuzmichov) [inline frontend/src/services/uploadFile.js:24]: constants for message

File: frontend/src/services/userService.js [hunk 8]
```diff
@@ -92,13 +92,13 @@ export const updateUserTeacher = (values) => {
...
+        .then(() => {
             getTeachersWithoutAccount();
             getUsersService();
             setLoadingService(false);
             successHandler(
-                i18n.t('serviceMessages:back_end_success_operation', {
-                    cardType: i18n.t('formElements:user_label'),
+                i18n.t(BACK_END_SUCCESS_OPERATION, {
+                    cardType: i18n.t(USER_LABEL),
                     actionType: i18n.t('serviceMessages:successfully_merged'),
```
Comment (andrii-kuzmichov) [inline frontend/src/services/userService.js:102]: maybe need constant for message also

File: frontend/src/validation/storeValidation.js [hunk 9]
```diff
@@ -1,21 +1,22 @@
...
-        find = store.getState().classActions.classScheduler.some(function (value, index, _arr) {
+        find = store.getState().classActions.classScheduler.some((value) => {
             return value.class_name === className && value.id !== classId;
         });
     } else {
-        find = store.getState().classActions.classScheduler.some(function (value, index, _arr) {
+        find = store.getState().classActions.classScheduler.some((value) => {
             return value.class_name === className;
         });
     }
```
Comment (andrii-kuzmichov) [inline frontend/src/validation/storeValidation.js:18]: need refactor, drop `if` with to `some` iteration

File: frontend/src/validation/storeValidation.js [hunk 10]
```diff
@@ -28,7 +29,7 @@ export const timeIntersectService = (startTime, endTime) => {
             moment(endTime, 'HH:mm').toDate(),
```
Comment (andrii-kuzmichov) [inline frontend/src/validation/storeValidation.js:29]: Create constants for time\date format

File: frontend/src/validation/storeValidation.js [hunk 11]
```diff
@@ -80,44 +81,44 @@ export const checkUniqueRoomName = (roomName) => {
...
-        find = store.getState().rooms.rooms.some(function (value, index, _arr) {
+        find = store.getState().rooms.rooms.some((value) => {
             return value.name.toUpperCase() === roomName.toUpperCase() && value.id !== roomdId;
         });
     } else {
-        find = store.getState().rooms.rooms.some(function (value, index, _arr) {
+        find = store.getState().rooms.rooms.some((value) => {
             return value.name.toUpperCase() === roomName.toUpperCase();
         });
     }
```
Comment (andrii-kuzmichov) [inline frontend/src/validation/storeValidation.js:91]: as above

File: frontend/src/validation/storeValidation.js [hunk 12]
```diff
@@ -80,44 +81,44 @@ export const checkUniqueRoomName = (roomName) => {
...
     if (!departmentTitle) {
-        return;
+        return undefined;
     }
-    const find = store.getState().departments.departments.some(function (value) {
+    const find = store.getState().departments.departments.some((value) => {
         return value.name.toUpperCase().trim() === departmentTitle.toUpperCase().trim();
     });
-    return find ? i18n.t('validationMessages:unique_error_message') : undefined;
+    return find ? i18n.t(UNIQUE_ERROR_MESSAGE) : undefined;
```
Comment (andrii-kuzmichov) [inline frontend/src/validation/storeValidation.js:121]: bad practice using `undefined`

**Review by andrii-kuzmichov — APPROVED** [review]

**Review by HiroFive — APPROVED** [review]

**Review by Akr1s — APPROVED** [review]

**Pull request merged**